### ISSUE TITLE
"Nginx::Request#headers_out" empties the assigned array, contrary to expectations

### ISSUE THREAD

**Issue by lamanotrama** [matsumoto-r/ngx_mruby#155]
I guess it is not the intended behavior.

Reproduction code is as follows.

```
# nginx

server {
  listen 127.0.0.1:8888;

  location / {
    mruby_rewrite_handler_code '
      headers = %w(abc=123 foo=bar)
      Nginx::Request.new.headers_out["Set-Cookies"] = headers
      Nginx.errlogger Nginx::LOG_ERR, headers
      Nginx::return headers.empty? ? Nginx::HTTP_INTERNAL_SERVER_ERROR : Nginx::OK
    ';
  }
}
```

Request to the above server as below.

```
$ curl http://127.0.0.1:8888 -sv > /dev/null
* Rebuilt URL to: http://127.0.0.1:8888/
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8888 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 127.0.0.1:8888
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
* Server nginx is not blacklisted
< Server: nginx
< Date: Thu, 04 Feb 2016 11:40:12 GMT
< Content-Type: text/html
< Content-Length: 186
< Connection: close
< Set-Cookies: foo=bar
< Set-Cookies: abc=123
<
{ [data not shown]
* Closing connection 0
root@node1:~#
```

and log

```
$ tail -1 /var/log/nginx/error.log
2016/02/04 11:40:12 [error] 11130#11130: *9 [], client: 127.0.0.1, server: , request: "GET / HTTP/1.1", host: "127.0.0.1:8888"
```

I expected status 200, and the array with two item in the log, but the results are not.

### Environment

 * Ubuntu 14.04.3
 * Nginx 1.9.9
 * nginx_mruby 1.15.0

For more details of nginx please refer https://gist.github.com/lamanotrama/638e97202cf54bfd9196

**Comment by matsumoto-r** [comment]
Thank you for your report. I'll investigate it.

**Comment by matsumoto-r** [comment]
Fixed via #156

**Comment by lamanotrama** [comment]
:100: 
Thank you!!
